test/test_transformer_metric_conversion.py

```diff
@@ -76,6 +81,102 @@ def test_mlkr(self):
+    # eigenvalues:
+    P = ortho_group.rvs(7, random_state=rng)
+
+    # matrix with all its coefficients very low (to check that the algorithm
+    # does not consider it as a diagonal matrix)(non regression test for
+    # https://github.com/metric-learn/metric-learn/issues/175)
+    M = np.diag([1e-15, 2e-16, 3e-15, 4e-16, 5e-15, 6e-16, 7e-15])
+    M = P.dot(M).dot(P.T)
+    L = transformer_from_metric(M)
+    assert_allclose(L.T.dot(L), M)
+
+    # diagonal matrix
+    M = np.diag(np.abs(rng.randn(5)))
+    L = transformer_from_metric(M)
+    assert_allclose(L.T.dot(L), M)
+
+    # low-rank matrix (with zeros)
+    M = np.zeros((7, 7))
+    small_random = rng.randn(3, 3)
+    M[:3, :3] = small_random.T.dot(small_random)
+    L = transformer_from_metric(M)
+    assert_allclose(L.T.dot(L), M)
+
+    # low-rank matrix (without necessarily zeros)
+    R = np.abs(rng.randn(7, 7))
+    M = R.dot(np.diag([1, 5, 3, 2, 0, 0, 0])).dot(R.T)
+    L = transformer_from_metric(M)
+    assert_allclose(L.T.dot(L), M)
+
+    # matrix with a determinant still high but which should be considered as a
+    # non-definite matrix (to check we don't test the definiteness with the
+    # determinant which is a bad strategy)
+    M = np.diag([1e5, 1e5, 1e5, 1e5, 1e5, 1e5, 1e-20])
+    M = P.dot(M).dot(P.T)
+    assert np.abs(np.linalg.det(M)) > 10
+    assert np.linalg.slogdet(M)[1] > 1  # (just to show that the computed
+    # determinant is far from null)
+    with pytest.raises(LinAlgError) as err_msg:
+      np.linalg.cholesky(M)
+    assert str(err_msg.value) == 'Matrix is not positive definite'
+    # (just to show that this case is indeed considered by numpy as an
+    # indefinite case)
+    L = transformer_from_metric(M)
+    assert_allclose(L.T.dot(L), M)
+
+    # matrix with lots of small nonzeros that make a big zero when multiplied
+    M = np.diag([1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3, 1e-3])
+    L = transformer_from_metric(M)
+    assert_allclose(L.T.dot(L), M)
+
+    # full rank matrix
+    M = rng.randn(10, 10)
+    M = M.T.dot(M)
+    assert np.linalg.matrix_rank(M) == 10
+    L = transformer_from_metric(M)
+    assert_allclose(L.T.dot(L), M)
+
+  def test_non_symmetric_matrix_raises(self):
+    """Checks that if a non symmetric matrix is given to
+    transformer_from_metric, an error is thrown"""
+    rng = np.random.RandomState(42)
+    M = rng.randn(10, 10)
+    with pytest.raises(ValueError) as raised_error:
+      transformer_from_metric(M)
+    assert str(raised_error.value) == "The input metric should be symmetric."
+
+  def test_non_psd_raises(self):
+    """Checks that a non PSD matrix (i.e. with negative eigenvalues) will
+    raise an error when passed to transformer_from_metric"""
+    rng = np.random.RandomState(42)
+    D = np.diag([1, 5, 3, 4.2, -4, -2, 1])
+    P = ortho_group.rvs(7, random_state=rng)
+    M = P.dot(D).dot(P.T)
+    with pytest.raises(ValueError) as raised_error:
+      transformer_from_metric(M)
```

Comment:
here you could also easily test that the error is raised in the diagonal case, for instance when calling `transformer_from_metric(D)`

Reply:
I agree, will do